data-pipeline/Cargo.toml

```diff
@@ -16,14 +16,16 @@ hyper = {version = "0.14", features = ["client"], default-features = false}
 log = "0.4"
 rmp-serde = "1.1.1"
 bytes = "1.4"
+either = "1.13.0"
```

Comment:
non-blocking: We can change this later, but is this really worth the overhead of adding a new crate (honest question)? Seems like just creating an Enum with two types would be sufficient? But perhaps there is some nuance I'm missing.

Reply:
Unfortunately it's not _just_ an enum with two types, it's also an enum with two types with some additional code to support "IntoIterator". For context, the reason I import this is that `tags` can be two different types depending on if some additional custom tags are provided, I don't expect `Either` (or any different "custom" enum we provide instead) to be used outside this context.

There's a couple different options here we can take but this seemed to be a pretty easy option to follow:
1. Keep it as is
2. Implement our own "Either" enum here including implementing [into_iter](https://docs.rs/either/latest/src/either/lib.rs.html#537) on it like the Either crate does
3. Remove the local variable and have some code duplication so no enum wrapping each type is required.

Reply:
I think that we should keep using the crate. Maybe I'm completely corrupted by my Scala days, but I think every language should have an `Either` type 😉